web-app/HeimScripts/heatmap/summary.R

```diff
@@ -98,9 +151,61 @@ check_input <- function(datasets, phase_info, projection)
   }
   
   
+
+
+
 }
 
 
+#preprocessed data contains one big data.frame. Summary stats and boxplot should be made per subset, so the dataset gets split on subsets
+split_on_subsets <- function(preprocessed_measurements)
+{
+  preprocessed_measurements <- preprocessed_measurements$preprocessed
+  
+  split_measurements <- list()
+  used_columns <- c()
+  
+  
+  # get possible subset identifiers. Format of column names for sample columns is <sample_name>_n<numerical node id>_s<numerical subset id>, eg. sample234_n1_s1
+  column_names <- colnames(preprocessed_measurements)
+
+  subsets_match_indices <- regexpr("s[[:digit:]]+$", column_names)
```

Comment:
as long as each subset is identified with a unique number, there's no problem for the script: it can handle it. So even if subsets get a longer identifier (for example one that is unique for this specific selected subset) it will still work. Will also work if it is for example decided to make >2 subsets possible.